snuba/query/parser/expressions.py

```diff
@@ -61,44 +62,70 @@ def visit_function_name(self, node: Node, visited_children: Iterable[Any]) -> st
     def visit_column_name(self, node: Node, visited_children: Iterable[Any]) -> Column:
         return Column(None, None, node.text)
 
-    def visit_empty(
-        self, node: Node, visited_children: Iterable[Any]
-    ) -> Optional[Expression]:
-        return None
+    def visit_low_pri_op(self, node: Node, visited_children: Iterable[Any]) -> Any:
+        return node.text
+
+    def visit_high_pri_op(self, node: Node, visited_children: Iterable[Any]) -> Any:
+        return node.text
+
+    def visit_arithmetic_term(
+        self, node: Node, visited_children: Tuple[Any, Expression, Any]
+    ) -> Expression:
+        _, term, _ = visited_children
+
+        return term
 
     def visit_low_pri_arithmetic(
         self,
         node: Node,
-        visited_children: Tuple[Any, Expression, Any, Tuple[str, Expression]],
+        visited_children: Tuple[
+            Any, Expression, Any, Union[Any, Tuple[str, Expression]]
+        ],
     ) -> Expression:
         _, term, _, exp = visited_children
 
-        if exp is None:
-            # A check for any None child nodes
-            # that arose from empty space ''
+        if exp == "":
             return term
-        else:
-            # return exp[1] because
-            # exp[0] is a string which
-            # represents the operator
-            return plus(term, exp[1])
+
+        for elem in exp:
+            if isinstance(elem, list):
+                if elem[0] == "+":
+                    term = plus(term, elem[1])
+                elif elem[0] == "-":
+                    term = minus(term, elem[1])
+            else:
+                if exp[0] == "+":
+                    return plus(term, exp[1])
+                elif exp[0] == "-":
+                    return minus(term, exp[1])
```

Comment:
I am not sure I get which case is this one.  When would you end up in this branch ?

Reply:
I added a comment in the updated code which (hopefully) briefly explains the purpose of having these two branches. Essentially, the two branches are for: 

1) when the (righthand) children are in the form of a List like: [ [operator, Expression], [operator, Expression], ... ], these need to be put together in proper FunctionCall() and need to be bubbled up. These Expressions can either be Literals or FunctionCall(). 

and 2) when the (righthand) children are in the form of [operator, Expression], where the Expression is already the correct FunctionCall() that has been bubbled up. In this case, we need only process the operator and put together another FunctionCall() (rather than iterating through a list).